# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/svg/types/scripted/SVGLength-cap.html
# Commit: 35034282dc7c
# Full Hash: 35034282dc7c8c0bcaf9552ee9fd226aa40a9769
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 1 - support rem, ch, ic and cap units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180618
# ==============================================================================

diff -r 06bbded14f7b -r 35034282dc7c testing/web-platform/tests/svg/types/scripted/SVGLength-cap.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/svg/types/scripted/SVGLength-cap.html	Mon Sep 11 18:20:18 2023 +0000
@@ -0,0 +1,23 @@
+<!DOCTYPE HTML>
+<title>SVGLength with 'cap' unit</title>
+<link rel="help" href="https://www.w3.org/TR/SVG/types.html#InterfaceSVGLength">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="cap_ref" style="font-family:initial; font-size:20px; width:10cap"></div>
+<svg>
+  <text id="cap_test" x="10cap" style="font-family:initial; font-size:20px"></text>
+</svg>
+<script>
+  let ref_width = cap_ref.offsetWidth;
+  let cap_length = cap_test.x.baseVal[0];
+
+  test(() => {
+    assert_equals(cap_length.unitType, SVGLength.SVG_LENGTHTYPE_UNKNOWN);
+    assert_approx_equals(cap_length.value, ref_width, 0.4);
+  }, "cap unit in SVGLength");
+
+  test(() => {
+    cap_length.value = ref_width * 2;
+    assert_approx_equals(cap_length.valueInSpecifiedUnits, 20, 0.1);
+  }, "Convert back to cap from new user unit value");
+</script>